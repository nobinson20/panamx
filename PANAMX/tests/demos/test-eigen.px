matrix eigen(matrix A) {
    double a;
    double b;
    double c;
    matrix vec;
    double e1;
    double e2;
    vec = <1, 2>;

    a = 1.0;
    b = -A[0][0] - A[1][1];
    c = A[0][0] * A[1][1] - A[0][1] * A[1][0];

    e1 = (-b+sqrtd((b*b-4.0*a*c))) / (2.0 * a);
    e2 = (-b-sqrtd((b*b-4.0*a*c))) / (2.0 * a);
    vec[0][0] = e1;
    vec[0][1] = e2;
    return vec;
}

int main() {
    matrix A;
    double a;
    double b;
    double c;
    matrix eigenvec;

    A = [3, 2;
         1, 4];
    eigenvec = eigen(A);

    prints("Matrix A:");
    printm(A);
    prints("Eigenvalues:");
    printm(eigenvec);
    return 0;
}
