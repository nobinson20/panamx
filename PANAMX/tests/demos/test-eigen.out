Matrix A:
[
	3.000   2.000
	1.000   4.000
]
Eigenvalues:
[
	5.000   2.000
]
